### PR TITLE
feat: add C implementation for `math/base/assert/is-nonpositive-finite`

### LINKED ISSUE #1831: [RFC]: add C implementation for `@stdlib/math/base/assert/is-nonpositive-finite`
### Description

This RFC proposes adding native C implementation for `math/base/assert/is-nonpositive-finite`:

```c
bool stdlib_base_is_nonpositive_finite( const double x );
```

### Related Issues

N/A

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
This commit if applied adds the C implementation to @stdlib/math/base/assert/is-nonpositive-finite package

## Description

This pull request:

-   adds the native C implementation to @stdlib/math/base/assert/is-nonpositive-finite

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1831 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2022 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md]:
```suggestion
Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 2]
```diff
@@ -35,10 +35,10 @@ var isNonPositiveFinite = require( '@stdlib/math/base/assert/is-nonpositive-fini
 Tests if a `numeric` value is a nonpositive finite number.
 
 ```javascript
-var bool = isNonPositiveFinite( 3.14 );
+var bool = isNonPositiveFinite( -3.14 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md:38]:
```suggestion
var bool = isNonPositiveFinite( 3.14 );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 3]
```diff
@@ -35,10 +35,10 @@ var isNonPositiveFinite = require( '@stdlib/math/base/assert/is-nonpositive-fini
...
 // returns true
 
-bool = isNonPositiveFinite( -2.0 );
+bool = isNonPositiveFinite( 2.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md:41]:
```suggestion
bool = isNonPositiveFinite( -2.0 );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 4]
```diff
@@ -61,13 +61,13 @@ bool = isNonPositiveFinite( -2.0 );
 ```javascript
 var isNonPositiveFinite = require( '@stdlib/math/base/assert/is-nonpositive-finite' );
 
-var bool = isNonPositiveFinite( 3.14 );
+var bool = isNonPositiveFinite( -3.14 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md:64]:
```suggestion
var bool = isNonPositiveFinite( 3.14 );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md (hunk 2 above)
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md:38]: thanks for catching this.

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 5]
```diff
@@ -81,6 +81,100 @@ bool = isNonPositiveFinite( NaN );
...
+
+int main( void ) {
+    double x;
+    bool v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+    x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
+        v = stdlib_base_is_nonpositive_finite( x );
+        printf( "x = %lf, is_nonpositive_finite(x) = %s\n", x, ( v ) ? "true" : "false" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md]:
```suggestion
        printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( v ) ? "True" : "False" );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 6]
```diff
@@ -81,6 +81,100 @@ bool = isNonPositiveFinite( NaN );
...
+    }
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+int main( void ) {
+	double x;
+	bool v;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
+		v = stdlib_base_is_nonpositive_finite( x );
+		printf( "x = %lf, is_nonpositive_finite(x) = %s\n", x, ( v ) ? "true" : "false" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c]:
```suggestion
		printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( v ) ? "True" : "False" );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/src/main.c [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+*/
+
+#include "stdlib/math/base/assert/is_nonpositive_finite.h"
+#include "stdlib/constants/float64/ninf.h"
+
+/**
+* Tests if a numeric value is a nonpositive finite number.
+*
+* @param x    input value
+* @return	  output value
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 9]
```diff
@@ -81,6 +81,99 @@ bool = isNonPositiveFinite( NaN );
...
+int main( void ) {
+    double x;
+    bool v;
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+    x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
+        v = stdlib_base_is_nonpositive_finite( x );
+        printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( v ) ? "True" : "False" );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md:164]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
        v = stdlib_base_is_nonpositive_finite( x );
        printf( "Value: %lf. Is NonPositive Finite? %s.\n", x, ( v ) ? "True" : "False" );
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 10]
```diff
@@ -81,6 +81,99 @@ bool = isNonPositiveFinite( NaN );
...
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md]:
```suggestion
    const double x[] = { 5.0, -5.0, 3.14, -3.14, 0.0, 0.0/0.0 };

    bool b;
    int i;
    for ( i = 0; i < 6; i++ ) {
        b = stdlib_base_is_nonpositive_finite( x[ i ] );
        printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( b ) ? "True" : "False" );
    }
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+int main( void ) {
+	double x;
+	bool v;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
+		v = stdlib_base_is_nonpositive_finite( x );
+		printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( v ) ? "True" : "False" );
+	}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c]:
```suggestion
	const double x[] = { 5.0, -5.0, 3.14, -3.14, 0.0, 0.0/0.0 };

	bool b;
	int i;
	for ( i = 0; i < 6; i++ ) {
		b = stdlib_base_is_nonpositive_finite( x[ i ] );
		printf( "Value: %lf. Is NonPositive Finite? %s.\n", x[ i ], ( b ) ? "True" : "False" );
	}
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nonpositive_finite.h"
+#include <stdbool.h>
+#include <stdlib.h>
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/examples/c/example.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md [hunk 13]
```diff
@@ -81,6 +81,99 @@ bool = isNonPositiveFinite( NaN );
...
+<!-- C API usage examples. -->
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/assert/is_nonpositive_finite.h"
+#include <stdbool.h>
+#include <stdlib.h>
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-nonpositive-finite/README.md]:
```suggestion
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @performant23 !

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**